### DISCUSSION TITLE
open image in new background tab?

### DISCUSSION THREAD

**Discussion by onguarde** [brookhong/Surfingkeys#1675] [Q&A]
We already have,

q: open button/images in current tab

I would a similar shortcut, but opening in background tab. 

Q: open multiple button/images in background tab 
(don't hide hints after each click, to allow multiple links to be clicked)

Any ideas, what to put in the settings?

**Comment by onguarde** [comment]
```
mapkey('Q', '#1Click on an Image or a button in background, multiple', function() {
    Hints.create("img, button", Hints.dispatchMouseClick, {multipleHits: true, tabbed: true, active: false});
});
```

This is what I came up with, with help from the sourcecode.

It seems to work, but the hints are NOT being displayed correctly. 


pressing q, (correct: expected hints should look like this, buttons and images only should have hints)
![image](https://user-images.githubusercontent.com/681355/155310166-e6dbda4f-1874-4a37-9ed5-d69969972ba1.png)


pressing Q, (wrong: images have no hints)
![image](https://user-images.githubusercontent.com/681355/155310215-9b93e6aa-ea90-4b90-b67d-0a0f9264ae63.png)

**Comment by onguarde** [comment]
@b0o any ideas?

**Comment by b0o** [comment]
> the hints are NOT being displayed correctly

They are being displayed correctly given the selector you're using. If you want them to be displayed differently, you'll need to adjust your selector. Try `.ymoOte a, .mJxzWe a img`.

Read up on [CSS Selectors](https://developer.mozilla.org/en-US/docs/Learn/CSS/Building_blocks/Selectors), the devtools [page inspector](https://developer.mozilla.org/en-US/docs/Tools/Page_Inspector/UI_Tour) (specifically the element picker), and [`document.querySelectorAll`](https://developer.mozilla.org/en-us/docs/web/api/document/queryselectorall) which you can use to test your selectors. Here's a [CSS Selectors video](https://www.youtube.com/watch?v=l1mER1bV0N0) from WebDevSimplified, haven't watched it myself but he usually makes good content. 

Tip: go to your devtools console and try to use `document.querySelectorAll` with the selector I shared: `document.querySelectorAll(".ymoOte a, .mJxzWe a img")`. 

![2022-02-23_22-34-47_firefoxdeveloperedition](https://user-images.githubusercontent.com/21299126/155471180-934a55b2-f87e-410d-91ea-7ac7092b5ac6.png)

If you're not getting the right elements, poke around in the page inspector and use the element picker to refine your selector. It's a process of trial and error.

**Reply by onguarde** [reply]
Thanks so much for the detailed reply. Will try this if i have time. 

I was thinking since "q" already creates the correct hints, it should already have those selectors in place. Is there a way to trigger the same function, but set an option to "open in background".

q: open button/images in current tab
Q: open multiple button/images in background tab
(don't hide hints after each click, to allow multiple links to be clicked)
